components/browser/session/src/main/java/mozilla/components/browser/session/SessionManager.kt

```diff
@@ -213,3 +213,23 @@ fun SessionManager.runWithSession(
     }
     return false
 }
+
+/**
+ * Tries to find a session with the provided session ID or uses them selectedSession and runs the block if found.
+ *
+ * @return True if the session was found and run successfully.
```

Comment:
It says it returns `True` if the session was found and the block run. But it returns what the block returns. So if the block returns `False` then even if the session was found and the block was run it returned false? Maybe that's correct and it's just the wording of this doc that is confusing?

Reply:
You are right. If the session was found, I was delegating to the lambda function to determine if the block run successfully. But the consumer just wants to know if the block was executed or not. I'll update it to reflect that 👍